README.md

```diff
@@ -64,6 +64,116 @@ where:
 
 * namespace: The namespace knative-eventing resources will be deployed into (it cannot be deployed into the same namespace as knative-operator or knative-serving
 
+## Usage with the Gateway APIs
+
+Kubernetes, Istio, and other service meshes are working toward a combined [Gateway API](https://gateway-api.sigs.k8s.io/) for Kubernetes Ingress, Load Balancing, and Service Mesh.  This would replace Istio's Gateway, VirtualService, and other resources.  There are currently two ways to use these APIs with Knative, both described below. 
+
+### Using the new Gateway API directly
+
+Knative's ingress/routing is configured by [net extensions](https://github.com/knative-extensions?q=net&type=all&language=&sort=).  We can replace the default [net-istio](https://github.com/knative-extensions/net-istio) with [net-gateway-api](https://github.com/knative-extensions/net-gateway-api) and Knative will natively generate/use the Gateway API resources.  In this example, we use the `istio-k8s` and `istio-ingress-k8s` charms, which support the new Gateway API, to deploy Istio:
+
+Deploy istio:
+```bash
+juju add-model istio-system
+juju deploy istio-k8s -m istio-system --channel edge --trust
+juju deploy istio-ingress-k8s -m istio-system --channel edge --trust
+```
+
+Deploy Knative:
+```bash
+juju add-model knative-user
+juju deploy ./charms/knative-operator/knative-operator_ubuntu-20.04-amd64.charm \
+  --model knative-user \
+  --trust \
+  --resource knative-operator-image=gcr.io/knative-releases/knative.dev/operator/cmd/operator@sha256:18e7a0c612efdce11d9d396fa9a1469fa206bdb2a817b5426a595ed57a5c3daf \
+  --resource knative-operator-webhook-image=gcr.io/knative-releases/knative.dev/operator/cmd/webhook@sha256:dc2b23c9f66869a8617a51f2f0af0f56d9dc6a85b71cff587161eb9711aef26a
+juju deploy ./charms/knative-serving/knative-serving_ubuntu-20.04-amd64.charm \
+  --model knative-user \
+  --trust \
+  --config namespace=knative-serving-1 \
+  --config istio.gateway.namespace=istio-system \
+  --config domain.name=10.64.140.43.nip.io \  # <-- may be different if you have a different load balancer configuration
```

Comment:
any idea why we need this, shouldn't the gateway name and namespace be enough since it already holds the LB data as it owns it?, what would happen if there's a discrepancy between the domain name used here and the External IP of the loadbalancer owned by the gateway?

Reply:
This is used to populate the links that knative tells you about.  It goes into the [config-domain](https://knative.dev/docs/serving/using-a-custom-domain/#configuring-domain-names) configmap.  So if you deploy a knative service called `my-knative-thing`, you can then do `kubectl get route my-knative-thing --output jsonpath="{.status.url}"` which would show `http://my-knative-thing.my-domain-name.com`.  the "my-domain-name.com" part comes from this `domain.name` config.  I think this is because we cannot really introspect the hostname based solely on the Gateway due to wildcard hosts

Which makes me wonder if we need to do something similar when using wildcards ourselves.  Maybe its just not feasible to introspect those, and instead a human needs to give guidance